src/java/org/apache/cassandra/index/sai/StorageAttachedIndex.java

```diff
@@ -497,11 +497,15 @@ public boolean isQueryable(Status status)
     }
 
     @Override
-    public void validate(PartitionUpdate update) throws InvalidRequestException
+    public void validate(PartitionUpdate update, ClientState state) throws InvalidRequestException
     {
         DecoratedKey key = update.partitionKey();
-        for (Row row : update)
-            validateMaxTermSizeForRow(key, row, true);
+
+        if (indexTermType.columnMetadata().isStatic())
```

Comment:
Do iiuc static rows were being missed before?

Reply:
As far as I can tell, they absolutely were missed.

Reply:
The static row isn't in the tree and so isn't returned from the iterator. Love static rows, huh?